File: rot13.cpp
#include <iostream>
#include <string>
#include <string.h>
#include <ctype.h>
using namespace std;

string rot13(char *input){
  //using ASCII decimal values
  string output;
  int len = strlen(input);
  for(int i=0; i<len; i++){
	if(isalpha(input[i]) ){
	  if(islower(input[i]) ){
		output += (input[i]+13)%('z'+1) < 'a' ? (input[i]+13)%('z'+1)+'a' : (input[i]+13)%('z'+1) ;
	  }
	  else {
		output += (input[i]+13)%('Z'+1) < 'A' ? (input[i]+13)%('Z'+1)+'A' : (input[i]+13)%('Z'+1);
	  }
	} else{
	  output+=input[i];
	}
  }
  return output;
}

int main(int argc, char** argv){
  if(argc<2){
	cerr<<"Usage:\t"<<argv[0]<<" \"<input text>\"";
	return 1;
  }
  cout<<argv[1]<<"\n"<<rot13(argv[1])<<"\n";
}

File: rot13.py
#!/usr/bin/python3
import sys 

def rot13(input):
    output=""
    for i in input:
        if i.isalpha():
            if i.islower():
                if (ord(i)+13)%ord('z') < ord('a'):
                    output += chr( ((ord(i)+13)%ord('z'))+ord('a') )
                else:
                    output += chr( (ord(i)+13)%ord('z') )
            else:
                if (ord(i)+13)%ord('Z') < ord('A'):
                    output += chr( ((ord(i)+13)%ord('Z'))+ord('A') )
                else:
                    output += chr( (ord(i)+13)%ord('Z') )
    return output

if len(sys.argv) < 2:
    print("Usage:\t %s \"<input text>\"" % sys.argv[0],  file=sys.stderr) #no can do, amigo
    sys.exit()

print( sys.argv[1], rot13(sys.argv[1]), sep="\n" )
